db.py

```diff
@@ -48,6 +51,16 @@ def allowed(self, access_level):
         else:
             return False
 
+    @staticmethod
+    def validate_unique_email(mapper, connect, target):
+        if target.email:
+            query = db.session.query(User).filter(User.email == target.email, User.id != target.id).exists()
+            if db.session.query(query).scalar():
```

Comment:
```suggestion
            if db.session.query(User).filter(User.email == target.email).first()
```
`target.id` is not set yet at this stage and is equals to `None`. The id is set after insertion as it is auto-incremental. By using `first()`, you will get the user or `None` if it does not exists, this is more direct.